pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go

```diff
@@ -119,28 +119,75 @@ func TestConfigMapControllerDisabling(t *testing.T) {
 		result              map[string][]string
 	}{
 		{
-			name: "ControllersEnabled",
+			name: "CapabilitiesEnabled",
 			knownCapabilities: []v1.ClusterVersionCapability{
 				configv1.ClusterVersionCapabilityBuild,
 				configv1.ClusterVersionCapabilityDeploymentConfig,
+				configv1.ClusterVersionCapabilityImageRegistry,
 			},
 			enabledCapabilities: []v1.ClusterVersionCapability{
 				configv1.ClusterVersionCapabilityBuild,
 				configv1.ClusterVersionCapabilityDeploymentConfig,
+				configv1.ClusterVersionCapabilityImageRegistry,
 			},
 			result: map[string][]string{"controllers": {"*"}},
 		},
 		{
-			name: "ControllersDisabled",
+			name: "BuildCapDisabled",
+			knownCapabilities: []v1.ClusterVersionCapability{
+				configv1.ClusterVersionCapabilityBuild,
+			},
+			enabledCapabilities: []v1.ClusterVersionCapability{},
+			result: map[string][]string{
+				"controllers": {"*",
+					"-openshift.io/build",
+					"-openshift.io/build-config-change",
+					"-openshift.io/builder-rolebindings",
+				}},
+		},
+		{
+			name: "DeploymentConfigCapDisabled",
+			knownCapabilities: []v1.ClusterVersionCapability{
+				configv1.ClusterVersionCapabilityDeploymentConfig,
+			},
+			enabledCapabilities: []v1.ClusterVersionCapability{},
+			result: map[string][]string{
+				"controllers": {"*",
+					"-openshift.io/deployer-rolebindings",
+					"-openshift.io/deploymentconfig",
+				}},
```

Comment:
/cc @ingvagabund  

Verify that these controllers should be disabled if DeploymentConfig is turned off.

Reply:
There's also `openshift.io/deployer` [1] that needs to be disabled. 
 [1] https://github.com/openshift/api/blob/88a8afea030c56055c81629b567fdcb5ab6a3903/openshiftcontrolplane/v1/types.go#L208